sdg_mapping/sdg_index/sdg_index_utils.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import pandas as pd
+
+import sys
+import os
+import json
+
+from sdg_mapping import project_dir
+
+with open(f'{project_dir}/data/aux/sdg_index_data_opts.json', 'r') as f:
+    state_dict = json.load(f)
+
+
+def sdg_index_file_path(year):
+    """sdg_index_file_path
+    To fetch filepath of SDG Index datasets.
+
+    Args:
+        year (str): Year of SDG Index datasets
+
+    Returns:
+        (str): File path
+
+    """
+    fname = f'{year}_sdg_index.xlsx'
+    return f'{project_dir}/data/raw/{fname}'
+
+
+def read_workbook(data_path, state):
+    """read_workbook
+    To read an Excel read_workbook
+
+    Args:
+        data_path (str): File path of an SDG Index dataset
+        state (str): State of dataset needed; raw or data
+
+    Returns:
+        (pd.DataFrame): Parsed SDG Index data
+
+    """
+    if state == 'data':
+        df = pd.read_excel(open(data_path, 'rb'),
+                  sheet_name=state_dict[state],header = 1)
+
+    elif state == 'raw':
+        df = pd.read_excel(open(data_path, 'rb'),
+                  sheet_name=state_dict[state])

```

Comment:
```suggestion
df = pd.read_excel(data_path, **read_opts)
```

Reply:
Also see the related changes to `data/aux/sdg_index_data_opts.json`

Reply:
What does `**read_opts` do? If I wanted the `raw` option, how would this look in this function?

Reply:
1. `read_opts = state_dict[state]` reads in the correct read options from `data/aux/sdg_index_data_opts.json` based on the `state` arg. `read_opts` is a dictionary.
2. `**read_opts` unpacks the dictionary into the keyword arguments for `pd.read_excel` where their names match keys in the dictionary so

```
df = pd.read_excel(data_path, **read_opts)

# is equivalent to

df = pd.read_excel(data_path, header=read_opts['header'], sheet_name=read_opts['sheet_name'])
```